src/classes/RD2_OpportunityService_TEST.cls

```diff
@@ -924,6 +924,120 @@ private with sharing class RD2_OpportunityService_TEST {
         assertPaymentIsMatchingOpp(oppById);
     }
 
+    /***
+    * @description Verifies if Account is not getting removed in opportunity
+    * when recurring donation record with only contact is updated.
+    */
+    @isTest
+    private static void shouldNotRemoveAccountInOppWhenRDWithNoAccountIsModified() {
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        setUpConfiguration();
+
+
+        npe03__Recurring_Donation__c rd = createRecurringDonation();
+
+        //create opp from rd with contact only
+        Opportunity opp = getOpportunityBuilderWithNoAccount(rd)
+            .withCloseDate(CLOSE_DATE)
+            .withOpenStage()
+            .build();
+        insert opp;
+
+
+        //Verify if accountid is populated in opp when only contact is provided in RD
+        List<Opportunity> opps = oppGateway.getRecords(rd);
+        System.assertEquals(1, opps.size(), 'One Installment Opp should be created on RD insert: ' + opps);
+        System.assert(opps[0].AccountId != null);
+
+        //update rd amount
+        Test.startTest();
+        rd.npe03__Amount__c = RD_NEW_AMOUNT;
+        update rd;
+        Test.stopTest();
+
+        //Verify if accountid is not nulled out when rd record is modified
+        List<Opportunity> oppUpdated = oppGateway.getRecords(rd);
+        System.assertEquals(RD_NEW_AMOUNT, oppUpdated[0].Amount, 'Amount should be updated');
+        System.assert(oppUpdated[0].AccountId != null);
+
+    }
+
+    /***
+    * @description Verifies if Account is not getting removed in opportunity
+    * when recurring donation record with account is modified.
+    */
+    private static void shouldNotRemoveAccountInOppWhenRDWithAccountIsModified() {
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        setUpConfiguration();
+
+        Account acc = UTIL_UnitTestData_TEST.buildOrganizationAccount();
+        insert acc;
+
+        npe03__Recurring_Donation__c rd = createRecurringDonation(acc);
+
+        //create opp from rd with account
+        Opportunity opp = getOpportunityBuilder(rd)
+            .withCloseDate(CLOSE_DATE)
+            .withOpenStage()
+            .build();
+        insert opp;
+
+        //update rd amount
+        Test.startTest();
+        rd.npe03__Amount__c = RD_NEW_AMOUNT;
+        update rd;
+        Test.stopTest();
+
+        //Verify if accountid is not changed or nulled out when rd record is modified
+        List<Opportunity> oppUpdated = oppGateway.getRecords(rd);
+        npe03__Recurring_Donation__c recurDonation = rdGateway.getRecord(rd.id);
```

Comment:
No need to create a new var for the RD. Based on the previous test comments, this modified test would look like:
```
/***
    * @description Verifies the Account on Opportunity is not nulled
    * when Recurring Donation having Organization specified is updated
    */
    private static void shouldNotRemoveAccountOnOppWhenRDWithAccountIsModified() {
        RD2_EnablementService_TEST.setRecurringDonations2Enabled();

        setUpConfiguration();

        Account acc = UTIL_UnitTestData_TEST.buildOrganizationAccount();
        insert acc;

        npe03__Recurring_Donation__c rd = createRecurringDonation(acc);

        Opportunity opp = getOpportunityBuilder(rd)
            .withCloseDate(CLOSE_DATE)
            .withOpenStage()
            .build();
        insert opp;

        Test.startTest();
        rd.npe03__Amount__c = RD_NEW_AMOUNT;
        update rd;
        Test.stopTest();

        rd = rdGateway.getRecord(rd.Id);
        List<Opportunity> opps = oppGateway.getRecords(rd);

        System.assertEquals(1, opps.size(), 'One Opp should exist: ' + opps);
        System.assertEquals(RD_NEW_AMOUNT, opps[0].Amount, 'Opp Amount should be updated');

        System.assertNotEquals(null, opps[0].AccountId, 
            'Opp Account should be specified when RD is updated');
        System.assertEquals(rd.npe03__Organization__c, opps[0].AccountId, 
            'Opp Account should be the same as the RD Account');
    }
```